### PR TITLE
added docs for header auth strategy

### PR DESCRIPTION
**Pull request by mlbiam**
for https://github.com/kiali/kiali/issues/3207

### CODE REVIEW

File: content/documentation/staging/configuration/authentication/_index.adoc [hunk 1]
```diff
@@ -9,6 +9,7 @@ hideIndex: true
 Kiali supports four authentication mechanisms:
 
 * link:{{< relref "anonymous" >}}[anonymous]: gives free access to Kiali.
+* link:{{< relref "header" >}}[header]: requires kiali to run behind a reverse proxy that is responsible for injecting the user's token or a toekn with impersonation.
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/_index.adoc:12]:
```suggestion
* link:{{< relref "header" >}}[header]: requires Kiali to run behind a reverse proxy that is responsible for injecting the user's token or a token with impersonation.
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Where `TOKEN` is the appropriate token for your cluster.  This `TOKEN` will be 
+submitted to the API server via a `TokenReview` to validate the token *ONLY*
+on the first access to Kiali.  On subsequent calls the `TOKEN` is passed through
+directly to the API server.
+
+== Security Considerations
+
+=== Network Polcies
+
+A policy should be put in place to make sure that the only "client" for kiali is
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/header.adoc:69]:
```suggestion
A policy should be put in place to make sure that the only "client" for Kiali is
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+the authenticating reverse proxy.  This helps limit potential abuse and ensures 
+that the authenticating reverse proxy is the source of truth for who accessed
+kiali.
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/header.adoc:72]:
```suggestion
Kiali.
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+=== Short Lived Tokens
+
+The authenticating reverse proxy should inject short lived token in the
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/header.adoc:76]:
```suggestion
The authenticating reverse proxy should inject a short lived token in the 
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+The authenticating reverse proxy should use the TokenRequest API instead of static 
+`ServiceAccount` tokens when possible while using impersonation. The 
+`ServiceAccount` that can impersonate users and groups is privileged and having it
+be short lived cuts down on the possibility of a token being leaked while it's being 
+passed between different parts of the infrastructure.
+
+==== Drop Incomming Impersonation Headers
+
+The authenticating proxy *MUST* drop any headers it receives from a remote client 
+that mach the impersonation headers.  Not only do you want to make sure that the
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/header.adoc:95]:
```suggestion
that match the impersonation headers.  Not only do you want to make sure that the
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+authenticating proxy can't be over-riden on which user to authenticate, but also
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/header.adoc:96]:
```suggestion
authenticating proxy can't be overriden on which user to authenticate, but also
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
minor typos and corrections

File: content/documentation/staging/configuration/authentication/_index.adoc [hunk 7]
```diff
@@ -9,6 +9,7 @@ hideIndex: true
 Kiali supports four authentication mechanisms:
```
Comment (jmazzitelli) [inline content/documentation/staging/configuration/authentication/_index.adoc:9]: There are now five (no longer four) of them:

```suggestion
Kiali supports five authentication mechanisms:
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Sorry - missed this one - the number of auth mechanisms has increased.

**Review by jmazzitelli — APPROVED** [review]
LGTM

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+----
+
+Where `TOKEN` is the appropriate token for your cluster.  This `TOKEN` will be 
+submitted to the API server via a `TokenReview` to validate the token *ONLY*
+on the first access to Kiali.  On subsequent calls the `TOKEN` is passed through
+directly to the API server.
+
+== Security Considerations
+
+=== Network Polcies
```
Comment (israel-hdez) [inline content/documentation/staging/configuration/authentication/header.adoc:67]: Fix to typo

```suggestion
=== Network Policies
```

File: content/documentation/staging/configuration/authentication/header.adoc [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+==== TokenRequest API
+
+The authenticating reverse proxy should use the TokenRequest API instead of static 
+`ServiceAccount` tokens when possible while using impersonation. The 
+`ServiceAccount` that can impersonate users and groups is privileged and having it
+be short lived cuts down on the possibility of a token being leaked while it's being 
+passed between different parts of the infrastructure.
+
+==== Drop Incomming Impersonation Headers
```
Comment (israel-hdez) [inline content/documentation/staging/configuration/authentication/header.adoc:92]: Another typo

```suggestion
==== Drop Incoming Impersonation Headers
```

**Pull request merged**